src/shader_recompiler/backend/spirv/spirv_emit_context.h

```diff
@@ -133,12 +144,109 @@ class EmitContext final : public Sirit::Module {
         return ConstantComposite(type, constituents);
     }
 
+    inline Id AddLabel() {
+        last_label = Module::AddLabel();
+        return last_label;
+    }
+
+    inline Id AddLabel(Id label) {
+        last_label = Module::AddLabel(label);
+        return last_label;
+    }
+
+    PointerType PointerTypeFromType(Id type) {
+        if (type.value == U8.value)
+            return PointerType::U8;
+        if (type.value == U16.value)
+            return PointerType::U16;
+        if (type.value == F16[1].value)
+            return PointerType::F16;
+        if (type.value == U32[1].value)
+            return PointerType::U32;
+        if (type.value == F32[1].value)
+            return PointerType::F32;
+        if (type.value == U64.value)
+            return PointerType::U64;
+        if (type.value == F64[1].value)
+            return PointerType::F64;
+        UNREACHABLE_MSG("Unknown type for pointer");
+    }
+
+    template <typename Func>
+    Id EmitMemoryAccess(Id type, Id address, Func&& fallback) {
```

Comment:
You should make this into a spirv function like the float conversion functions. A shader might have a dozen S_LOAD_DWORD instructions, so inlining this code will bloat the shader